src/Controllers/SchemaController.js

```diff
@@ -1096,9 +1096,16 @@ export default class SchemaController {
     maintenance?: boolean
   ) {
     if (fieldName.indexOf('.') > 0) {
-      // subdocument key (x.y) => ok if x is of type 'object'
-      fieldName = fieldName.split('.')[0];
-      type = 'Object';
+      // "<array>.<index>" for Nested Arrays
+      // "<embedded document>.<field>" for Nested Objects
+      // JSON Arrays are treated as Nested Objects
+      const [x, y] = fieldName.split('.');
+      fieldName = x;
+      if (!isNaN(y) && !['sentPerUTCOffset', 'failedPerUTCOffset'].includes(fieldName)) {
```

Comment:
- Hard-coded field names look suspicious. If certain fields require special treatment for this logic to work we may end up with the same issue as in https://github.com/parse-community/Parse-SDK-JS/issues/2198.

- I believe we should be stricter than `isNaN`. Without using regex (too expensive) the only solution I can think of is on a per char basis; not sure about its perf though:

  ```
  const isArrayIndex = Array.from(str).every(char => char >= '0' && char <= '9')
  ```

Reply:
I added the stricter array index check.

The hard coded field names come from the `_PushStatus` schema. They have array indexes but are saved as objects instead of arrays.

Reply:
> They have array indexes but are saved as objects instead of arrays.

What does that mean? Is this the issue discussed in https://github.com/parse-community/Parse-SDK-JS/issues/2198#issuecomment-2208635633 where we don't know whether an object key `0` is an array index or a normal object key?

Reply:
The tests pass, that's a good sign. What would we have to change for the push status so we don't have to treat them differently here?

Reply:
Pretty much nothing we can do here. These are built into Parse instead of created by the user.

If you want to see how it works check out this test

https://github.com/parse-community/parse-server/blob/alpha/spec/PushWorker.spec.js#L324

Reply:
I see, this seems to be a Parse internal syntax where the key name contains a value, separated by a dot, correct? In that case I believe it makes sense.